core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessage.java

```diff
@@ -165,6 +176,20 @@ protected EventExecutor defaultSubscriberExecutor() {
         return completionFuture;
     }
 
+    @Nullable
+    @Override
+    public Throwable completionCause() {
+        return completionCauseSupplier != null ? completionCauseSupplier.get() : null;
```

Comment:
Could memoize the completion cause so it’s instantiated once. Please make sure that this method is used when creating a `CloseEvent`, too.